src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -776,15 +926,30 @@ export class ShadowGenerator implements IShadowGenerator {
      * @param mapSize The size of the texture what stores the shadows. Example : 1024.
      * @param light The light object generating the shadows.
      * @param usefulFloatFirst By default the generator will try to use half float textures but if you need precision (for self shadowing for instance), you can use this option to enforce full float texture.
+	 * @param cascades Number of cascades for CSM. Values between MIN_CASCADES and MAX_CASCADES.
+	 * @param csmFrustumLength The distance of the furthest frustum of the camera for CSM.
      */
-    constructor(mapSize: number, light: IShadowLight, usefulFloatFirst?: boolean) {
+    constructor(mapSize: number, light: IShadowLight, usefulFloatFirst?: boolean, cascades?: number, csmFrustumLength?: number) {
         this._mapSize = mapSize;
         this._light = light;
         this._scene = light.getScene();
         light._shadowGenerator = this;
 
         ShadowGenerator._SceneComponentInitialization(this._scene);
 
+        // sets cascade value and the csmFrustumLength if any, valid only for directional lights

```

Comment:
This could be done in the rebuild function to ensure that if the type of shadow changes we would do the same

Reply:
ok, pulled out into a separate ```private _checkAndInitCascades``` function.